pkg/plugins/runtime/k8s/controllers/inbound_converter.go

```diff
@@ -124,7 +124,9 @@ func InboundInterfacesFor(zone string, pod *kube_core.Pod, services []*kube_core
 func InboundTagsForService(zone string, pod *kube_core.Pod, svc *kube_core.Service, svcPort *kube_core.ServicePort) map[string]string {
 	tags := util_k8s.CopyStringMap(pod.Labels)
 	for key, value := range tags {
-		if value == "" {
+		// we don't want to convert labels like
+		// kuma.io/sidecar-injection, kuma.io/service, k8s.kuma.io/namespace etc.
+		if strings.Contains(key, "kuma.io/") || value == "" {
```

Comment:
That's a good step to avoiding accidental override. Would it make sense to reject or at least log messages when it's set? (except `kuma.io/sidecar-injection`)

Reply:
Should we use `strings.HasPrefix`? We have to reflect this limitation in docs, it may sound strange if we say `your label can't contain "kuma.io/" substring`. I think it's better to write about limitation in terms of prefixes `reserved prefixes are "kuma.io/" and "k8s.kuma.io/"`. WDYT?

Reply:
+1 about making both kuma.io and k8s.kuma.io reserved rather than contains!

Reply:
there is also `traffic.kuma.io/`, `prometheus.metrics.kuma.io/`, but ok. We can do explicit prefixes